### PR TITLE
Enable Arm64 and arm build

### PR DESCRIPTION
**Pull request by bingbing8**
Enable Arm64 and arm build

### CODE REVIEW

File: contrib/win32/win32compat/shell-host.c [hunk 1]
```diff
@@ -1652,26 +1652,30 @@ static void* xmalloc(size_t size) {
...
-
-	SET_USER_ENV(FOLDERID_LocalAppData, L"LOCALAPPDATA");
-	SET_USER_ENV(FOLDERID_Profile, L"USERPROFILE");
-	SET_USER_ENV(FOLDERID_RoamingAppData, L"APPDATA");
+	DWORD len = GetCurrentDirectory(_countof(userprofile_path), userprofile_path);
+	if (len > 0) {
+		SetEnvironmentVariableW(L"USERPROFILE", userprofile_path);
+		swprintf_s(path, _countof(path), L"%s\\AppData\\Local", userprofile_path);
+		SetEnvironmentVariableW(L"LOCALAPPDATA", path);
+		swprintf_s(path, _countof(path), L"%s\\AppData\Roaming", userprofile_path);
```
Comment (bagajjal) [inline contrib/win32/win32compat/shell-host.c]: \\AppData\\Roaming? (\ is missing before \Roaming)

File: channels.h [hunk 2]
```diff
@@ -1,339 +1,339 @@
...
-void	 chan_rcvd_oclose(struct ssh *, Channel *);
-void	 chan_rcvd_eow(struct ssh *, Channel *);
-void	 chan_read_failed(struct ssh *, Channel *);
-void	 chan_ibuf_empty(struct ssh *, Channel *);
-void	 chan_rcvd_ieof(struct ssh *, Channel *);
-void	 chan_write_failed(struct ssh *, Channel *);
-void	 chan_obuf_empty(struct ssh *, Channel *);
-
-#endif
+/* $OpenBSD: channels.h,v 1.130 2017/09/21 19:16:53 markus Exp $ */
```
Comment (bagajjal) [inline channels.h:1]: hard to find what has changed in this file..

Comment (bingbing8) [reply channels.h:1]: this file was included by mistake. nothing change in this file

File: contrib/win32/win32compat/shell-host.c (hunk 1 above)
Comment (bingbing8) [reply contrib/win32/win32compat/shell-host.c]: fixed

**Pull request closed without merging**